src/pages/LiquidityProvider/index.tsx

```diff
@@ -51,14 +51,16 @@ export const refreshProviders = async (
 ): Promise<void> => {
   if (network === 'liquid') {
     try {
-      dispatch(clearProviders());
-      dispatch(clearMarkets());
       const providersFromRegistry = await getProvidersFromTDexRegistry();
       const currentEndpoints = providers.map((provider) => provider.endpoint);
       const newProviders = providersFromRegistry.filter((p) => !currentEndpoints.includes(p.endpoint));
-      const actions = newProviders.map(addProvider);
-      const toastSuccessAction = addSuccessToast(`${actions.length} new providers from TDEX registry!`);
-      [...actions, toastSuccessAction].forEach(dispatch);
+      if (newProviders.length) {
```

Comment:
is it safe? ie. >0? or Array.isArray?

Reply:
filter() just above always returns an array, so I think it is safe.